### PR TITLE
feat: add proxy-setup.sh

### PR DESCRIPTION
**Pull request by bap2pecs**
## Summary

i am following https://www.notion.so/snapchain/op-devnet-deployment-125a62189ff480abb8e0c7bb86db92bf to set up DNS and SSL but there are many steps so it's error-prone.

adding automation steps

## Test Plan

set env vars and run `sudo make dns-setup`

then check success
```
$ echo $?
0
```

also see

image

now verify the certs are valid for the 3 domains:

```
$ sudo openssl x509 -in /etc/letsencrypt/live/tohma.snapchain.dev/fullchain.pem -text | grep DNS:
                DNS:demo.tohma.snapchain.dev, DNS:finality-rpc.tohma.snapchain.dev, DNS:finality.tohma.snapchain.dev
```

modify the env var:
```
NEXT_PUBLIC_FINALITY_GADGET_API_URL=https://finality-rpc.tohma.snapchain.dev
```

and restart the finality explorer

now verify the finality gadget rpc:

```
curl https://finality-rpc.tohma.snapchain.dev/v1/chainSyncStatus
{"latest_block":118191,"latest_btc_finalized_block":118180,"earliest_btc_finalized_block":66709,"latest_eth_finalized_block":117989}
```

also https://finality.tohma.snapchain.dev/ and https://demo.tohma.snapchain.dev/ loads

test demo app rpc:

```
$ curl https://demo.tohma.snapchain.dev/api/getTransactions/0x00000000080480b4dee2a7fe4e59d7ce1289d83c
[]%
```

### CODE REVIEW

**Review by lesterli — APPROVED** [review]

File: configs/nginx/demo-app.conf.template [hunk 1]
```diff
@@ -0,0 +1,29 @@
+server {
+    listen 443 ssl;
+    server_name demo.${CERTBOT_DOMAIN_SUFFIX};
+
+    ssl_certificate /etc/letsencrypt/live/${CERTBOT_DOMAIN_SUFFIX}/fullchain.pem;
+    ssl_certificate_key /etc/letsencrypt/live/${CERTBOT_DOMAIN_SUFFIX}/privkey.pem;
+
+    # Backend API
+    location /api/ {
```
Comment (bap2pecs) [inline configs/nginx/demo-app.conf.template:9]: @parketh jfyi this mean the .env file needs to be changed locally

File: .env.babylon-integration.example [hunk 2]
```diff
@@ -64,4 +64,20 @@ STAKING_TIME=10000 # ~70 days
 STAKING_AMOUNT=10000 # 0.0001 BTC
 
 # Finality Explorer Configuration
-NEXT_PUBLIC_FINALITY_GADGET_API_URL=http://11.22.33.44:18080
\ No newline at end of file
+NEXT_PUBLIC_FINALITY_GADGET_API_URL=http://11.22.33.44:18080
+
+# DNS Configuration
+FINALITY_SYSTEM_SERVER_IP=11.22.33.44
```
Comment (parketh) [inline .env.babylon-integration.example:70]: do we need to add another PR to devops to set these env vars?

Comment (parketh) [reply .env.babylon-integration.example:70]: btw, since we clone this example file to create the .env, it may be better to leave these values unset so as to not break the existing script. if deployment doesnt need SSL, it should work without setting these values

File: Makefile [hunk 3]
```diff
@@ -101,3 +101,7 @@ toggle-cw-killswitch:
 teardown:
 	@./scripts/babylon-integration/teardown.sh
 .PHONY: teardown
+
+proxy-setup:
```
Comment (parketh) [inline Makefile:105]: maybe we can add another make command to restart nginx server (for debugging)

File: configs/nginx/demo-app.conf.template (hunk 1 above)
Comment (parketh) [reply configs/nginx/demo-app.conf.template:9]: wdym? as in the .env file for existing deployments?

File: scripts/babylon-integration/proxy-setup.sh [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+# the certs will be stored in /etc/letsencrypt/live/${CERTBOT_DOMAIN_SUFFIX}
+# 
+# note that Certbot creates a single certificate that's valid for all those 
+# domains (called a SAN - Subject Alternative Names certificate)
+# 
+# after running the command, you can verify by:
+#   sudo openssl x509 -in /etc/letsencrypt/live/${CERTBOT_DOMAIN_SUFFIX}/fullchain.pem -text | grep DNS:
+# 
+# reference: https://eff-certbot.readthedocs.io/en/latest/using.html
+sudo certbot certonly --nginx --non-interactive --agree-tos -m ${CERTBOT_EMAIL} \
```
Comment (parketh) [inline scripts/babylon-integration/proxy-setup.sh:50]: interesting. is it also possible to renew the certificate for new subdomains we add in future?

File: configs/nginx/demo-app.conf.template (hunk 1 above)
Comment (bap2pecs) [reply configs/nginx/demo-app.conf.template:9]: sry i mean you need to change the `packages/frontend/.env` on your local machine to have 

```
NEXT_PUBLIC_BACKEND_API_HOST=https://localhost:8000/api
```

File: .env.babylon-integration.example (hunk 2 above)
Comment (bap2pecs) [reply .env.babylon-integration.example:70]: yes we could add it to devops

how will these value break the existing script if they are not used in those scripts?

if the deployment doesn't need SSL, these vars won't be used?

File: Makefile (hunk 3 above)
Comment (bap2pecs) [reply Makefile:105]: that's just `sudo systemctl reload nginx.service`. feel too small to be it's own make command

File: .env.babylon-integration.example (hunk 2 above)
Comment (bap2pecs) [reply .env.babylon-integration.example:70]: ah got it. let me remove <> then

Comment (bap2pecs) [reply .env.babylon-integration.example:70]: pushed the fix to main
